packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart

```diff
@@ -411,23 +431,9 @@ class FirebaseAuth extends FirebasePluginPlatform {
         forceRecaptchaFlow: forceRecaptchaFlow);
   }
 
-  /// Changes the current type of persistence on the current Auth instance for
-  /// the currently saved Auth session and applies this type of persistence for
-  /// future sign-in requests, including sign-in with redirect requests.
-  ///
-  /// This will return a promise that will resolve once the state finishes
-  /// copying from one type of storage to the other. Calling a sign-in method
-  /// after changing persistence will wait for that persistence change to
-  /// complete before applying it on the new Auth state.
-  ///
-  /// This makes it easy for a user signing in to specify whether their session
-  /// should be remembered or not. It also makes it easier to never persist the
-  /// Auth state for applications that are shared by other users or have
-  /// sensitive data.
-  ///
-  /// This is only supported on web based platforms.
   Future<void> setPersistence(Persistence persistence) async {
```

Comment:
Can we add the `Deprecated` annotation here so developers will get notified in their IDE?

Reply:
Also, do we need to breaking change this straight away? Is it not possible to deprecate but just call the new method instead, and then eventually we can remove the method in a later release.

Reply:
@Ehesp this is useless now `Firebase.instance.setPersistence()` because the value needs to be passed before `Firebase.instance` is ever called. We could just deprecate it and keep the implementation but it does nothing. We needed a new API (`FirebaseAuth.persistenceType()`) to pass the value on the very first call of `initializeAuth()` for web. We felt it made sense to simply remove underlying implementation and throw a deprecation exception so users adopt the new strategy.

Reply:
I've updated the PR description, but just to be clear, `setPersistence()` is not useless as long as persistence is used to initialize the auth instance [here](https://github.com/firebase/flutterfire/pull/9138/files#diff-073c6ae89d1947a587a79d2020e2cd9478a0d218a4aeb8dd6bce4b37344b92e1R23-R52)